Crud api

AleksandarSavic95 opened the pull request.

AleksandarSavic95 committed:
#ALSA-23 setup jwt-auth + 2 users in database

- users added using a seeder

*note:* valid login request doesn't generate the jwt token

AleksandarSavic95 committed:
#ALSA-24 setup CORS

TODO: be strict with origin settings in cors.php

AleksandarSavic95 committed:
#ALSA-14 make TodoItem's model

- added Priority constants (low, mid, high),
- tested obtaining a constant value in web.php :)

AleksandarSavic95 committed:
#ALSA-14 finish crud api

GET request to http://127.0.0.1:8000/api/todoitems/42 returns `[]`

AleksandarSavic95 committed:
#ALSA-14 change variable names in TodoItemController, test with Postman

+ Successfully tested CRUD API with Postman

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+{
+    public function index()
+    {
+        return TodoItem::all();
```
Je l' si razmisljao da ti mozda ne trebaju bas svi ToDo-ovi i od svih korisnika? :D Moras po ID-ju user-a uzimati sta ti treba..ovo ce uvek vracati svacije.

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+
+    public function store(Request $request)
+    {
+        // \Log::info($request);
```
Visak :D

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+
+        $todoitem = TodoItem::create($request->all());
+
+        return response()->json($todoitem, 201);
```
Ako se ne varam, ovo ti je visak. Laravel ce da ti handle-a konverziju u `.json`. 
Dovoljno je da kazes
`return $todoItem;`

Ili jos bolje, oneliner
`return TodoItem::create($request->all());`

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+    {
+        // \Log::info($request);
+
+        $todoitem = TodoItem::create($request->all());
```
Koristi `camelCase`

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+    {
+        $todoitem->update($request->all());
+
+        return response()->json($todoitem, 200);
```
Isto vazi kao i gore. :D

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+        return response()->json($todoitem, 200);
+    }
+
+    public function delete(TodoItem $todoitem)
```
Obicno se za `delete` i `update` kao drugi parametar prima `id`, ne ceo item.

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+    {
+        $todoitem->delete();
+
+        return response()->json(null, 204);
```
Isto kao i gore. :D

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -0,0 +1,42 @@
+        return response()->json($todoitem, 201);
+    }
+
+    public function update(Request $request, TodoItem $todoitem)
```
Napisala sam ispod, ovde bi trebalo da primis `id`. :D

AleksandarSavic95 replied:
https://www.toptal.com/laravel/restful-laravel-api-tutorial
Implicit route model binding :))

AleksandarSavic95 committed:
#ALSA-14 fix todo_items table

- change `content` type to `text` (longer than `string`'s 255 chars)
- fixed indents from tabs to 4 spaces
- installed `doctrine/dbal` for DB column modification

AleksandarSavic95 committed:
#ALSA-14 make user-todo_items connection

Tested in `api.php` with the `/users/1/todoitems` route

AleksandarSavic95 committed:
#ALSA-14 fix reviewed CRUD api, merge with branch "model"

- add camelCase where missing,
- `user_id` is:
    - inserted from JWT in `TodoItemController@store` method,
    - removed from $request in `TodoItemController@update` method
- one step closer to jwt integration

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -4,38 +4,45 @@
     {
-        return TodoItem::all();
+        $userId = 1;
+        // $userId = Auth::User()->id; // Auth::user() ?
```
Komentare brisiiii

calico-anne commented on app/Http/Controllers/TodoItemController.php:
```diff
@@ -4,38 +4,45 @@
-        return response()->json($todoitem, 201);
+        // TODO: set user_id of created item to id of the authenticated user
+        // array_merge is used because it overrides the existing key!
+        return TodoItem::create(array_merge($request->all(), ['user_id' => 1]));
```
`['user_id' => Auth::id()]`

AleksandarSavic95 committed:
#ALSA-14 remove comments
